src/main/java/gift/jwt/OAuthMethodArgumentResolver.java

```diff
@@ -0,0 +1,34 @@
+package gift.jwt;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+@Component
+public class OAuthMethodArgumentResolver implements HandlerMethodArgumentResolver {
+    private final JwtService jwtService;
+
+    public OAuthMethodArgumentResolver(JwtService jwtService) {
+        this.jwtService = jwtService;
+    }
+
+    @Override
+    public boolean supportsParameter(MethodParameter parameter) {
+        boolean hasOAuthAnnotation = parameter.hasParameterAnnotation(OAuth.class);
```

Comment:
저는 이런거 그냥 inline 시켜버리는게 깔끔하다고 생각합니다ㅎㅎ (취향차이라서 반영 안하셔도 되어요!)
```suggestion
        return parameter.hasParameterAnnotation(OAuth.class);
```